zulipterminal/ui_tools/views.py

```diff
@@ -775,6 +777,55 @@ def __init__(self, controller: Any, widgets: List[Any],
         self.log = urwid.SimpleFocusListWalker(widgets)
         super().__init__(self.log)
 
+    def max_widths(self, contents: PopUpViewContent, title_len: int,
+                   dividechars: int) -> List[int]:
+        """
+        Returns maximum width required per column and sets width accordingly.
+        """
+        # Add +6 to compensate for the surrounding borders.
+        heading_width = title_len + 6
+        row_widths = []
+        for category, content in contents:
+            heading_width = max(len(category), heading_width)
+            for row in content:
+                if isinstance(row, str):
+                    heading_width = max(len(row), heading_width)
+                elif isinstance(row, tuple):
+                    row_widths.append(list(map(len, row)))
+        col_max_widths = [max(row_width) for row_width in zip(*row_widths)]
+        self.width = max(heading_width, sum(col_max_widths) + dividechars)
+        return col_max_widths
+
+    def build_widgets(self, contents: PopUpViewContent, title_len: int,
```

Comment:
I know that `title_len` is what is really passed in, but this value need not be the length of the title - what does it mean to pass in a certain value, for the purposes of these functions? Does that give you an idea for a better name?

Reply:
There are corner cases wherein the calculated width for the 'content' is smaller than the popup title. I use `title_len` to make sure that the calculated width is wide enough to display the popup title every time.

I will rename this to something better.